howdoi/howdoi.py

```diff
@@ -66,22 +65,40 @@ def u(x):
     'bing': SCHEME + 'www.bing.com/search?q=site:{0}%20{1}',
     'google': SCHEME + 'www.google.com/search?q=site:{0}%20{1}'
 }
+
+BLOCK_INDICATORS = (
+        'form id="captcha-form"',
+        'This page appears when Google automatically detects requests coming from your computer network which appear to be in violation of the <a href="//www.google.com/policies/terms/">Terms of Service'
+        )
+
 STAR_HEADER = u('\u2605')
 ANSWER_HEADER = u('{2}  Answer from {0} {2}\n{1}')
 NO_ANSWER_MSG = '< no answer given >'
-XDG_CACHE_DIR = os.environ.get('XDG_CACHE_HOME',
-                               os.path.join(os.path.expanduser('~'), '.cache'))
-CACHE_DIR = os.path.join(XDG_CACHE_DIR, 'howdoi')
-CACHE_FILE = os.path.join(CACHE_DIR, 'cache{0}'.format(
-    sys.version_info[0] if sys.version_info[0] == 3 else ''))
+
+CACHE_DIR = appdirs.user_cache_dir('howdoi')
+CACHE_ENTRY_MAX = 128
 
 if os.getenv('HOWDOI_DISABLE_CACHE'):
-    howdoi_session = requests.session()
+    cache = NullCache() # works like an always empty cache, cleaner than 'if cache:' everywhere
 else:
-    if not os.path.exists(CACHE_DIR):
-        os.makedirs(CACHE_DIR)
-    howdoi_session = requests_cache.CachedSession(CACHE_FILE)
+    cache = FileSystemCache(CACHE_DIR, CACHE_ENTRY_MAX, default_timeout=0)
+
+howdoi_session = requests.session()
+
+class BlockError(Exception):
+    pass
+
+def _random_int(width):
```

Comment:
Why the move away from `random`?

Reply:
Small win. Easy to do, minimal code, 0 portability issues.

Reply:
I don't think that generating random numbers or selecting random elements should be re-written in this repo. In a bit a research I did see that `random` isn't cryptographically secure and thus you might want to use `urandom`, but unless there are serious portability or speed considerations with `random` then let's use the library.

Reply:
Yes, there's a few ms win here from removing the import, more on a non-ssd machine with a cold cache. It's what, 10 lines of perfectly straightforward code? I went through the whole import profile and looked if anything could be jettisoned without much trouble. This was the easiest one. It's bad to re-implement for no reason, but I'm lazy and always have a reason if I'm going to do something ;)

Reply:
It's essentially the equivalent of inlining something for perf purposes in C. I'll remove it if you absolutely insist, but I don't find the arguments against doing this compelling.